pkg/util/log/audit/audit.go

```diff
@@ -63,128 +71,139 @@ var (
-		payload.EnvCloudRole = v
+		h.payload.EnvCloudRole = v
 		delete(entry.Data, EnvKeyCloudRole)
 	}
 
 	if v, ok := entry.Data[EnvKeyCloudRoleVer].(string); ok {
-		payload.EnvCloudRoleVer = v
+		h.payload.EnvCloudRoleVer = v
 		delete(entry.Data, EnvKeyCloudRoleVer)
 	}
 
-	payload.EnvCloudRoleInstance = h.env.Hostname()
-	payload.EnvCloudEnvironment = h.env.Environment().Name
-	payload.EnvCloudLocation = h.env.Location()
+	if v, ok := entry.Data[EnvKeyHostname].(string); ok {
+		h.payload.EnvCloudRoleInstance = v
+		delete(entry.Data, EnvKeyHostname)
+	}
+
+	if v, ok := entry.Data[EnvKeyLocation].(string); ok {
+		h.payload.EnvCloudLocation = v
+		delete(entry.Data, EnvKeyLocation)
+	}
 
 	if v, ok := entry.Data[EnvKeyCloudDeploymentUnit].(string); ok {
-		payload.EnvCloudDeploymentUnit = v
+		h.payload.EnvCloudDeploymentUnit = v
 		delete(entry.Data, EnvKeyCloudDeploymentUnit)
 	}
 
-	payload.EnvCloudVer = ifxAuditCloudVer
+	h.payload.EnvCloudVer = IFXAuditCloudVer
 
 	// Part-B
-	if ids, ok := entry.Data[PayloadKeyCallerIdentities].([]*CallerIdentity); ok {
-		payload.CallerIdentities = append(payload.CallerIdentities, ids...)
+	if ids, ok := entry.Data[PayloadKeyCallerIdentities].([]CallerIdentity); ok {
+		h.payload.CallerIdentities = append(h.payload.CallerIdentities, ids...)
 		delete(entry.Data, PayloadKeyCallerIdentities)
 	}
 
 	if v, ok := entry.Data[PayloadKeyCategory].(string); ok {
-		payload.Category = Category(v)
+		h.payload.Category = v
 		delete(entry.Data, PayloadKeyCategory)
 	}
 
 	if v, ok := entry.Data[PayloadKeyOperationName].(string); ok {
-		payload.OperationName = v
+		h.payload.OperationName = v
 		delete(entry.Data, PayloadKeyOperationName)
 	}
 
-	if v, ok := entry.Data[PayloadKeyResult].(*Result); ok {
-		payload.Result = v
+	if v, ok := entry.Data[PayloadKeyResult].(Result); ok {
+		h.payload.Result = v
 		delete(entry.Data, PayloadKeyResult)
 	}
 
 	if v, ok := entry.Data[PayloadKeyRequestID].(string); ok {
-		payload.RequestID = v
+		h.payload.RequestID = v
 		delete(entry.Data, PayloadKeyRequestID)
 	}
 
-	if rs, ok := entry.Data[PayloadKeyTargetResources].([]*TargetResource); ok {
-		payload.TargetResources = append(payload.TargetResources, rs...)
+	if rs, ok := entry.Data[PayloadKeyTargetResources].([]TargetResource); ok {
+		h.payload.TargetResources = append(h.payload.TargetResources, rs...)
 		delete(entry.Data, PayloadKeyTargetResources)
 	}
 
 	// add the audit payload
-	b, err := json.Marshal(payload)
+	b, err := json.Marshal(h.payload)
 	if err != nil {
 		return err
 	}
 	entry.Data[MetadataPayload] = string(b)
 
 	// add non-IFxAudit metadata for our own use
 	entry.Data[MetadataCreatedTime] = logTime
-	entry.Data[MetadataLogKind] = ifxAuditLogKind
+	entry.Data[MetadataLogKind] = IFXAuditLogKind
+	entry.Data[MetadataSource] = h.payload.EnvAppID
 
+	adminOp := false
+	if strings.Contains(h.payload.OperationName, "/admin") {
+		adminOp = true
+	}
+	entry.Data[MetadataAdminOperation] = adminOp
```

Comment:
this test should be strengthened and should be moved to pkg/frontend/middleware/log.go.  There is already a test `if vars["api-version"] == admin.APIVersion || strings.HasPrefix(r.URL.Path, "/admin")` in there - make that a helper function and reuse it